pkg/iterator/emitters.go

```diff
@@ -76,7 +79,82 @@ func (e *TableEmitter) SnapshotMetadataIteratorRecord(recordNumber int, metadata
 		fmt.Fprintf(e.Writer, tableRowFmt, recordNumber, metadata.VolumeCapacityBytes, bmt, bmd.ByteOffset, bmd.SizeBytes)
 	}
 
-	return true
+	return nil
+}
+
+func (e *TableEmitter) SnapshotMetadataIteratorDone(_ int) error {
+	return nil
+}
+
+// VerifierEmitter formats the metadata as a table.
+type VerifierEmitter struct {
+	// SourceDevice contains the source device file descriptor.
+	SourceDevice *os.File
+
+	// TargetDevice contains the target device file descriptor.
+	TargetDevice *os.File
+}
+
+func (verifierEmitter *VerifierEmitter) SnapshotMetadataIteratorRecord(_ int, metadata IteratorMetadata) error {
+	for _, bmd := range metadata.BlockMetadata {
+		buffer := make([]byte, bmd.SizeBytes)
```

Comment:
Why allocate the buffer in the inner loop? This is essentially serialized code and the buffer can be reused.

In fact, you could even stash the buffer in the `VerifyEmitter` struct itself, as a private variable that gets initialized before use if `nil`.

Reply:
> Why allocate the buffer in the inner loop? This is essentially serialized code and the buffer can be reused.
> 
> In fact, you could even stash the buffer in the `VerifyEmitter` struct itself, as a private variable that gets initialized before use if `nil`.

SizeBytes is only available in the loop ?
buffer of required size can only be created in the loop ?

Reply:
Good point. But then this logic would be very vulnerable to extent based responses as it is blindly allocating a buffer.... imagine if you had a single range for a 1 Ti disk!

Probably better to invoke a subroutine that can split the I/O if the size exceeds some maximum.